src/Database/Type/DateTimeType.php

```diff
@@ -157,7 +157,7 @@ public function marshal($value)
             if ($value === '' || $value === null || $value === false || $value === true) {
                 return null;
             }
-            if (is_numeric($value)) {
+            if (ctype_digit($value)) {
```

Comment:
Our tests don't complain about any type handling issues, so I don't know what problem you're seeing.